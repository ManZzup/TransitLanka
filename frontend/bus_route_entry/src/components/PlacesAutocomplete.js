import React from 'react';

const PlacesAutocomplete = (props) => (
    <div className="small-12 large-5 columns">
      <div className="input-group">
        <span className="input-group-label">{props.label}</span>
        <input className="input-group-field" id={props.auto_input_id} type="text" disabled={props.disabled}
                onChange={props.onSearchInputChange} value={props.searchTxt} />
      </div>
    </div>
);

export default PlacesAutocomplete;
