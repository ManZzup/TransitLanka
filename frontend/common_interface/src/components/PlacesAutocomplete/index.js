import React from 'react';

const PlacesAutocomplete = (props) => {
return(
    <div className="row">
      <div className="input-group">
        <span className="input-group-label">{props.label}</span>
        <input className="input-group-field awesomplete" type="text"
               placeholder={props.placeholder}
               id={props.id}
               onChange={(e) => props.onTextChange(e.target.value)}  />
      </div>
    </div>
);
}

export default PlacesAutocomplete;
