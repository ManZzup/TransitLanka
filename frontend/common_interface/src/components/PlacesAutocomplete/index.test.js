import React from 'react';
import PlacesAutocomplete from './';
import renderer from 'react-test-renderer';

it('renders autocomplete text input correctly', () => {
  const tree = renderer.create(
    <PlacesAutocomplete label="Test Input" placeholder="Test Input" defaultValue="Test Input" id="0" />
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
