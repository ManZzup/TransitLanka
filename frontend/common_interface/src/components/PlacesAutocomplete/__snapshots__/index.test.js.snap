exports[`test renders autocomplete text input correctly 1`] = `
<div
  className="row">
  <div
    className="input-group">
    <span
      className="input-group-label">
      Test Input
    </span>
    <input
      className="input-group-field awesomplete"
      defaultValue={undefined}
      id="0"
      onChange={[Function]}
      placeholder="Test Input"
      type="text" />
  </div>
</div>
`;
